llmfoundry/utils/builders.py

```diff
@@ -407,6 +407,10 @@ def build_tokenizer(
             int(1e30),
         )
 
+    if not hasattr(tokenizer, 'eos_token_id') or tokenizer.eos_token_id is None:
+        raise ValueError(
+            f'The tokenizer {tokenizer_name} must have an eos_token_id.')
+
```

Comment:
let's actually check for `eos_token` instead of `eos_token_id`.

`eos_token_id` uses `eos_token` to get an id, so `eos_token` is a better "source of truth"